sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -322,6 +323,62 @@ class Analyzer(
   }
 
   /**
+    * Replace the index with the related attribute for order by and group by.
+    */
+  object ResolveIndexReferences extends Rule[LogicalPlan] {
```

Comment:
What's your preference? It's not a must to be a new Rule. How about embedded it into `ResolvedReferences`? I was thinking of putting this similar logic into one place.